### ISSUE TITLE
Ldapadvanced bugs and user logins

### ISSUE THREAD

**Issue by timlegge** [kimai/kimai#946]
#### Expected behaviour
Users login and sometimes provide a username in the wrong format:
example\username
username@example.com

the ldap authenticator handles it by stripping the prefix and suffix.  Ldapadvanced does not

also special characters in the distinguished name are not escaped and will prevent login

#### Actual behaviour
User cannot login in eithe case of wrong format or special characters
#### Steps to reproduce
Setup ldap and pass the username in the wrong format 
Have a dn that includes special characters

#### Reproducable on demo.kimai.org? [yes|no]
It would be with ldapadvanced against a open ldap or active directory server

#### Kimai version [for example: dev 1.1.0.1389|github master|github develop]
Kimai 1.1.0.1389 
#### Relevant log entries from log file (debug tab or temporary/logfile.txt)

For the special characters:
E_WARNING ldap_search(): Search: Bad search filter @/var/www/kimai_1.1.0/libraries/Kimai/Auth/Ldapadvanced.php line 333

I have preliminary fixes that I need to clean up that work in our environment.  I will send a pull request with better implementations for this:

        $prefix = 'example\\';
        $suffix = '@example.com';

       if (substr(strtoupper($username), 0, strlen($prefix)) == strtoupper($prefix)) {
           $username = substr($username, strlen($prefix));
       }
       if (substr(strtoupper($username), strlen($username)-strlen($suffix), strlen($username)) == strtoupper($suffix)) {
           $username = substr($username, 0, strlen($username) - strlen($suffix));
       }

and:

        $distinguishedName = str_replace('(' , '\28' , $distinguishedName );
        $distinguishedName = str_replace(')' , '\29' , $distinguishedName );

**Comment by marquesmatheus** [comment]
I'm having trouble with ldapadvanced authentication.
I've created a php sample to connect to my LDAP and it goes well.
I have a scenario that my AD don't need to have bindDN or bindPW, but i've changed everything and tested everything and still get the error on "PHP Warning:  ldap_search(): Search: Operations error in /var/www/kimai-ts/libraries/Kimai/Auth/Ldapadvanced.php on line 256"...
Almost giving up ... :(
I've tested comment the bind part of this file, it points to my AD but it ignore the password, i can connect using my local network ad username and with any kind of password.
Tried to edit directly on file, created the auth.php and changed the autoconf.php, but no sucess to connect using ldap, either ldap or ldapadvanced. 
Need serious help on connecting and using ldapadvanced.


Thx
